### DISCUSSION TITLE
cammini di peso minimo 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#35] [Esercizi A-L]
Sia $G$ un grafo non diretto, e sia $w : E(G) \to \mathbb{R}^+$ che associa ad ogni arco un peso. Sia $P$ un cammino di peso minimo tra due vertici $x, y \in V(G)$ 

Definiamo $w' : E(G) \to \mathbb{R}^+ : (u, v) \to 2 \cdot w(u, v)$

$P$ è ancora un cammino di peso minimo per $w'$? Giustificare la risposta.

**Comment by IonutCicio** [comment]
Si, $P$ rimane un cammino di peso minimo

### Dimostrazione

#### Osservazione

Sia $Q$ un qualsiasi cammino in $G$, si ha che $w'(Q) = 2w(Q)$. Infatti, per definizione si ha

$$w'(Q) = \sum\limits_{e \in Q} w'(e) =  \sum\limits_{e \in Q} 2w(e) = 2\sum\limits_{e \in Q} w(e) = 2w(Q)$$

#### Assurdo

Supponiamo per assurdo che $\exists Q$ cammino di $G$ tale che $w'(Q) < w'(P) \implies 2w(Q) < 2w(P) \implies w(Q) < w(P)$ che è una contraddizione, dato che $P$ era il cammino con peso $w$ minimo.

**Comment by interestingmaneuver** [comment]
Questa volta provo più formalmente (rispetto al n.1):

Con abuso di notazione, definisco $w$ e $w'$ anche per i cammini:
$$w'(P) = \sum_{(u,v) \in P} 2w(u,v)$$

Si vede subito che $w'(P) = 2w(P)$. Allora, se esiste $P'$ tale che $w'(P') \le w'(P)$, necessariamente si ha $w(P') \le w(P)$ (uguale se e solo se era uguale in $w'$). Siccome $P$ era il cammino minimo secondo $w$, si ha che $w'(P) = w'(P')$.

Quindi $P$ è il cammino minimo anche secondo $w'$.
